gef.py

```diff
@@ -560,6 +565,127 @@ def __init__(self, elf="", minimalist=False):
             self.e_shentsize, self.e_shnum, self.e_shstrndx = struct.unpack("{}HHH".format(endian), fd.read(6))
         return
 
+    def get_machine_name(self):
+        return {
+            0x02: "SPARC",
+            0x03: "X86",
+            0x08: "MIPS",
+            0x14: "PowerPC",
+            0x15: "PowerPC64",
+            0x28: "ARM",
+            0x2b: "SPARC64",
+            0x3e: "X86_64",
+            0xb7: "AARCH64",
+            0xf3: "RISCV",
+            None: None
+        }[self.e_machine]
+
+
+class PE:
+    """Basic PE parsing.
+    This class is not complete, please extend it further.
+    """
+    X86_64              = 0x8664
+    X86_32              = 0x14c
+    ARM                 = 0x1c0
+    ARM64               = 0xaa64
+    ARMNT               = 0x1c4
+    AM33                = 0x1d3
+    IA64                = 0x200
+    EFI                 = 0xebc
+    MIPS                = 0x166
+    MIPS16              = 0x266
+    MIPSFPU             = 0x366
+    MIPSFPU16           = 0x466
+    WCEMIPSV2           = 0x169
+    POWERPC             = 0x1f0
+    POWERPCFP           = 0x1f1
+    SH3                 = 0x1a2
+    SH3DSP              = 0x1a3
+    SH4                 = 0x1a6
+    SH5                 = 0x1a8
+    THUMP               = 0x1c2
+    RISCV32             = 0x5032
+    RISCV64             = 0x5064
+    RISCV128            = 0x5128
+    M32R                = 0x9041
+
+    dos_magic           = b'MZ'
+    ptr_to_pe_header    = None
+    pe_magic            = b'PE'
+    machine             = X86_32
+    num_of_sections     = None
+    size_of_opt_header  = None
+    dll_charac          = None
+    opt_magic           = b'\x02\x0b'
+    entry_point         = None
+    base_of_code        = None
+    image_base          = None
+
+
+    def __init__(self, pe=""):
+        if not os.access(pe, os.R_OK):
+            err("'{0}' not found/readable".format(pe))
+            err("Failed to get file debug information, most of gef features will not work")
+            return
+
+        with open(pe, "rb") as fd:
+            # off 0x0
+            self.dos_magic = fd.read(2)
+            if self.dos_magic != PE.dos_magic:
+                self.machine = None
+                return
+
+            fd.read(0x3a)
```

Comment:
Might be nicer to just make a `struct.Struct` and parse out all the relevant fields in one go. It would be more description as well.

Either way, we should probably seek to get around, `fd.seek(ABS_OFFSET, 0)` or `fd.seek(OFFSET, 1)